pyccel/codegen/printing/ccode.py

```diff
@@ -470,17 +470,17 @@ def _print_Literal(self, expr):
 
     def _print_LiteralComplex(self, expr):
         if expr.real == LiteralFloat(0):
-            return self._print(PyccelAssociativeParenthesis(PyccelMul(expr.imag, LiteralImaginaryUnit())))
+            return self._print(PyccelAssociativeParenthesis(PyccelMul(expr.imag, LiteralImaginaryUnit(), simplify = True)))
         else:
             return self._print(PyccelAssociativeParenthesis(PyccelAdd(expr.real,
-                            PyccelMul(expr.imag, LiteralImaginaryUnit()))))
+                            PyccelMul(expr.imag, LiteralImaginaryUnit(), simplify = True))))
```

Comment:
This shouldn't be simplified. You cannot get simpler than a LiteralComplex so you end up with an infinite loop.
E.g: consider `LiteralComplex(1,2)`
Here `expr=LiteralComplex(1,2)`
We want to use `PyccelAssociativeParenthesis`, `PyccelMul`, and `LiteralImaginaryUnit` to print it as `(1+2j)` but this expression is then simpilified to `LiteralComplex(1,2)` which we try to print using this function and so on